### PR TITLE
load sigar using SigarUtil

### PR DESCRIPTION
**Pull request by nishantmonu51**
fix loading of native sigar lib for JvmCpuMonitor.

### CODE REVIEW

File: src/main/java/com/metamx/metrics/SigarUtil.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        File nativeLibTmpFile = new File(tmpDir, libName);
+        nativeLibTmpFile.deleteOnExit();
+        StreamUtils.copyToFileAndClose(url.openStream(), nativeLibTmpFile);
+        log.info("Loading sigar native lib at tmpPath[%s]", nativeLibTmpFile);
+        loader.load(nativeLibTmpFile.getParent());
+      } else {
+        log.info("No native libs found in jar, letting the normal load mechanisms figger it out.");
+      }
+    }
+    catch (Exception e) {
```
Comment (drcrallen) [inline src/main/java/com/metamx/metrics/SigarUtil.java]: can this be targeted exceptions? Things like InterruptedException require special handling which is lost in blanket catches like this.

File: src/test/java/com/metamx/metrics/SigarLoadTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.hyperic.sigar.SigarException;
+import org.junit.Test;
+
+public class SigarLoadTest
+{
+  @Test
+  public void testSigarLoad() throws SigarException
+  {
+    Sigar sigar = SigarUtil.getSigar();
+    Assert.assertEquals(sigar.getCpuInfoList().length, Runtime.getRuntime().availableProcessors());
```
Comment (drcrallen) [inline src/test/java/com/metamx/metrics/SigarLoadTest.java]: Do these both draw from the same information? example: if I have a cpuset limited cgroup, will they both show the number of total cores, or will they only look at the number of cores available in the cpuset?

Comment (xvrl) [inline src/test/java/com/metamx/metrics/SigarLoadTest.java]: we can just test that it's not empty, assuming the user running the test always has the permissions to get that information, which may not always be true.

Maybe a better test is to find something that any user will always have access to, regardless of permissions and test that?

Comment (xvrl) [inline src/test/java/com/metamx/metrics/SigarLoadTest.java]: `getPid()` should be pretty safe, and check it's > 0?

File: src/main/java/com/metamx/metrics/SigarUtil.java (hunk 1 above)
Comment (nishantmonu51) [inline src/main/java/com/metamx/metrics/SigarUtil.java]: done.

File: src/test/java/com/metamx/metrics/SigarLoadTest.java (hunk 2 above)
Comment (nishantmonu51) [inline src/test/java/com/metamx/metrics/SigarLoadTest.java]: modified to check for pid.

**Pull request merged**